muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -1,71 +1,315 @@
+import fi.otavanopisto.muikku.users.WorkspaceUserEntityController;
 import fi.otavanopisto.security.rest.RESTPermit;
 import fi.otavanopisto.security.rest.RESTPermit.Handling;
 
 @Path("/records")
 @RequestScoped
 @Stateful
-@Produces("*/*")
+@Produces("application/json")
 @RestCatchSchoolDataExceptions
 public class TranscriptofRecordsRESTService extends PluginRESTService {
 
   private static final long serialVersionUID = 1L;
-  
+  private static final int MAX_COURSE_NUMBER = 15;
+
   @Inject
   private TranscriptOfRecordsFileController transcriptOfRecordsFileController;
-  
+
   @Inject
   private SessionController sessionController;
-  
+
+  @Inject
+  private WorkspaceController workspaceController;
+
+  @Inject
+  private CourseMetaController courseMetaController;
+
+  @Inject
+  private TranscriptOfRecordsController vopsController;
+
+  @Inject
+  private UserController userController;
+
+  @Inject
+  private WorkspaceUserEntityController workspaceUserEntityController;
+
+  @Inject
+  private EnvironmentUserController environmentUserController;
+
+  @Inject
+  private PluginSettingsController pluginSettingsController;
+
+  @Inject
+  private GradingController gradingController;
+
+  @Inject
+  private Logger logger;
+
   @GET
   @Path("/files/{ID}/content")
   @RESTPermit(handling = Handling.INLINE)
+  @Produces("*/*")
   public Response getFileContent(@PathParam("ID") Long fileId) {
-    
+
     if (!sessionController.isLoggedIn()) {
       return Response.status(Status.FORBIDDEN).entity("Must be logged in").build();
     }
-    
+
     UserEntity loggedUserEntity = sessionController.getLoggedUserEntity();
 
     TranscriptOfRecordsFile file = transcriptOfRecordsFileController
         .findFileById(fileId);
-    
+
     if (file == null) {
       return Response.status(Status.NOT_FOUND).entity("File not found").build();
     }
 
     boolean isLoggedUser = Objects.equals(file.getUserEntityId(), loggedUserEntity.getId());
-    
+
     if (!isLoggedUser) {
       return Response.status(Status.FORBIDDEN).entity("Not your file").build();
     }
-    
+
     StreamingOutput output = s -> transcriptOfRecordsFileController.outputFileToStream(file, s);
-    
+
     String contentType = file.getContentType();
-    
+
     return Response.ok().type(contentType).entity(output).build();
   }
-  
+
+  @GET
+  @Path("/vops/{IDENTIFIER}")
+  @RESTPermit(handling = Handling.INLINE)
+  public Response getVops(@PathParam("IDENTIFIER") String studentIdentifierString) {
+
+    String educationTypeMappingString = pluginSettingsController.getPluginSetting("transcriptofrecords", "educationTypeMapping");
+    EducationTypeMapping educationTypeMapping = new EducationTypeMapping();
+    if (educationTypeMappingString != null) {
+      try {
+        educationTypeMapping = new ObjectMapper().readValue(educationTypeMappingString, EducationTypeMapping.class);
+      } catch (IOException e) {
+        logger.log(Level.INFO, "Education type mapping not set");
```

Comment:
Should be returned as a bad request as a null educationTypeMapping would result in an NPE later on